File: 2025-09/2025-09-03/pangram.js
function isPangram(sentence, letters) {
  sentence = sentence.replace(/[^a-zA-Z]/g, "").toLowerCase();

  return isItPangram(sentence, letters) && isItPangram(letters, sentence);
}

function isItPangram(string1, string2) {
  for (let i = 0; i < string2.length; i ++) {
    if (string1.indexOf(string2[i]) === -1)
      return false;
  }
  return true;
}

console.log(isPangram("hello", "helo"));
console.log(isPangram("hello", "hel"));
console.log(isPangram("hello", "helow"));
console.log(isPangram("hello world", "helowrd"));
console.log(isPangram("Hello World!", "helowrd"));
console.log(isPangram("Hello World!", "heliowrd"));
console.log(isPangram("freeCodeCamp", "frcdmp"));
console.log(isPangram("The quick brown fox jumps over the lazy dog.", "abcdefghijklmnopqrstuvwxyz"));

File: 2025-09/2025-09-03/pangram.py
import re

def is_pangram(sentence, letters):
    regex = re.compile(r'[^a-zA-Z]')
    sentence = re.sub(regex, "", sentence).lower()
    
    return is_it_pangram(sentence, letters) and is_it_pangram(letters, sentence)

def is_it_pangram(string1, string2):
    for character in string2:
        if not character in string1:
            return False
    return True

print(is_pangram("hello", "helo"))
print(is_pangram("hello", "hel"))
print(is_pangram("hello", "helow"))
print(is_pangram("hello world", "helowrd"))
print(is_pangram("Hello World!", "helowrd"))
print(is_pangram("Hello World!", "heliowrd"))
print(is_pangram("freeCodeCamp", "frcdmp"))
print(is_pangram("The quick brown fox jumps over the lazy dog.", "abcdefghijklmnopqrstuvwxyz"))